In [1]:
# Add the src directory to the Python path
import sys
import pandas as pd
sys.path.append('../src')
# Import necessary modules
from hypothesis_testing import HypothesisTester

In [2]:

# Load the processed data
cleaned_data = pd.read_csv('../data/Processed/cleaned_data.csv', low_memory=False)


In [3]:
tester = HypothesisTester(cleaned_data)

### Risk Differences Across Provinces
#### Analysis Objective
This test examines whether there are significant differences in risk levels (measured by Total Claims) across provinces. The goal is to understand how risk varies regionally, which can inform province-specific policies or risk management strategies.
### Hypothesis
Null Hypothesis (H₀): No risk differences across provinces.

Alternative Hypothesis (H₁): Risk differences exist across provinces.

In [8]:

# 1. Test risk differences across provinces
province_result = tester.test_risk_by_group('Province', 'TotalClaims')
print(province_result)

Test: ANOVA
Statistic: 5.849413762407606
P-value: 1.6782057588675903e-07
Message: Reject Null Hypothesis: There are significant differences.
None


#### Results

- Test Type: ANOVA (Analysis of Variance)
  - Compares the mean Total Claims across multiple provinces.
- F-Statistic: 5.849
  - Indicates significant variation in Total Claims between provinces.
- p-Value: 1.6782e-07
 - This value is much smaller than the common significance level of 0.05 indicating that the observed differences are statistically significant.
- Decision: Reject the null hypothesis (H₀).
#### Conclusion
There is sufficient evidence to conclude that significant risk differences exist across provinces. The variations in Total Claims are statistically significant and not due to random chance.
#### Implications
1. Targeted Risk Strategy: Risk management or pricing strategies may need to be adjusted by province.
2. Further Exploration: Identify specific provinces driving the differences and tailor policies accordingly.
3. Potential Policy Adjustments: Review how premiums and policies can be aligned with the provincial risk profiles.
#### Next Steps
Visualize Total Claims by province to pinpoint specific regional disparities.

Investigate other factors contributing to the risk differences between provinces.

## Risk Differences Between Zip Codes

### Analysis Objective
This test evaluates whether there are significant differences in **risk levels** (measured by `Total Claims`) between zip codes. The goal is to determine if zip code-specific risk management or pricing strategies are warranted.

### Hypotheses
- **Null Hypothesis (H₀):** No risk differences between zip codes.
- **Alternative Hypothesis (H₁):** Risk differences exist between zip codes.

In [9]:
# 2. Test risk differences between zip codes
zip_result = tester.test_risk_by_group('PostalCode', 'TotalClaims')
print(zip_result)


Test: ANOVA
Statistic: 0.9419762214391849
P-value: 0.8906511279164051
Message: Accept Null Hypothesis: There are no significant differences.
None


### Results
- **Test Type:** ANOVA (Analysis of Variance)  
  - Compares the mean `Total Claims` across multiple zip codes.
- **F-Statistic:** 0.942  
  - Indicates a very small variance in `Total Claims` between zip codes.
- **p-Value:** 0.891  
  - This p-value is much greater than the common significance level of 0.05, suggesting that the observed differences are likely due to random chance.
- **Decision:** Accept the null hypothesis (H₀).

### Conclusion
There is **insufficient evidence** to reject the null hypothesis, meaning there are no significant risk differences between zip codes. Any variations in `Total Claims` across zip codes are likely due to random fluctuation rather than any systematic difference.

### Implications
1. **Uniform Risk Strategy:** Since no significant differences exist between zip codes, it may not be necessary to implement zip code-specific risk management or pricing strategies.
2. **Resource Allocation:** The same risk management approach can be applied uniformly across zip codes, as there is no compelling evidence to suggest otherwise.
3. **Exploring Other Factors:** Further analysis could explore whether other variables, such as demographics or policy types, influence risk within zip codes.
